fineract-provider/src/main/java/org/apache/fineract/cob/loan/CapitalizedIncomeAmortizationBusinessStep.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.fineract.infrastructure.event.business.service.BusinessEventNotifierService;
+import org.apache.fineract.organisation.monetary.domain.Money;
+import org.apache.fineract.organisation.monetary.domain.MoneyHelper;
+import org.apache.fineract.portfolio.loanaccount.data.AccountingBridgeDataDTO;
+import org.apache.fineract.portfolio.loanaccount.data.AccountingBridgeLoanTransactionDTO;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanCapitalizedIncomeBalance;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTransaction;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTransactionRepository;
+import org.apache.fineract.portfolio.loanaccount.mapper.LoanAccountingBridgeMapper;
+import org.apache.fineract.portfolio.loanaccount.repository.LoanCapitalizedIncomeBalanceRepository;
+import org.apache.fineract.portfolio.loanaccount.util.CapitalizedIncomeAmortizationUtil;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class CapitalizedIncomeAmortizationBusinessStep implements LoanCOBBusinessStep {
+
+    private final LoanCapitalizedIncomeBalanceRepository capitalizedIncomeBalanceRepository;
+    private final BusinessEventNotifierService businessEventNotifierService;
+    private final JournalEntryWritePlatformService journalEntryWritePlatformService;
+    private final LoanAccountingBridgeMapper loanAccountingBridgeMapper;
+    private final LoanTransactionRepository loanTransactionRepository;
+    private final ExternalIdFactory externalIdFactory;
+
+    @Transactional
+    @Override
+    public Loan execute(Loan loan) {
+        MathContext mc = MoneyHelper.getMathContext();
+        LocalDate businessDate = DateUtils.getBusinessLocalDate();
+        List<LoanCapitalizedIncomeBalance> balances = capitalizedIncomeBalanceRepository.findAllByLoanId(loan.getId());
+
+        BigDecimal totalAmortizationAmount = BigDecimal.ZERO;
+        for (LoanCapitalizedIncomeBalance balance : balances) {
+            long daysUntilMaturity = businessDate.until(loan.getMaturityDate(), ChronoUnit.DAYS);
+            Money dailyAmortization = CapitalizedIncomeAmortizationUtil.calculateDailyAmortization(
+                    loan.getLoanProductRelatedDetail().getCapitalizedIncomeStrategy(), daysUntilMaturity, balance.getUnrecognizedAmount(),
+                    loan.getCurrency().toData());
+            totalAmortizationAmount = totalAmortizationAmount.add(dailyAmortization.getAmount(), mc);
+            balance.setUnrecognizedAmount(balance.getUnrecognizedAmount().subtract(dailyAmortization.getAmount(), mc));
+        }
+
+        capitalizedIncomeBalanceRepository.saveAll(balances);
+
+        if (MathUtil.isGreaterThanZero(totalAmortizationAmount)) {
+            LoanTransaction transaction = LoanTransaction.capitalizedIncomeAmortization(loan,
```

Comment:
Money.of(loan.getCurrency(), transactionAmount) is enough.
Also please make sure the fee or interest portion is set based on the capitalized income type

Reply:
LoanTransaction.capitalizedIncomeAmortization() handles the portions based on capitalized income type.